src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -1595,5 +1600,56 @@ private static void VerifyDerivedTypeConstraints(IEdmModel model, IEdmVocabulary
 
             throw new ODataException(Strings.PathParser_TypeCastOnlyAllowedInDerivedTypeConstraint(fullTypeName, kind, name));
         }
+
+        private static string ResolveEscapeFunction(string identifier, IEdmType bindingType, IEdmModel model, out string parenthesisExpression)
+        {
+            Debug.Assert(identifier != null && identifier.Length >= 1 && identifier[0] == ':');
+            Debug.Assert(bindingType != null);
+
+            bool isComposableRequired = identifier.Length >= 2 && identifier[identifier.Length - 1] == ':';
+            IEnumerable<IEdmFunction> functions = model.FindBoundOperations(bindingType).OfType<IEdmFunction>().Where(f => IsUrlEscapeFunction(model, f));
+
+            IEdmFunction function = functions.FirstOrDefault(f => f.IsComposable == isComposableRequired);
+            if (function == null)
+            {
+                // If no non-composable escape function exist, use the composable function
+                if (!isComposableRequired)

```

Comment:
Should we just make this part of the previous condition? i.e., if (function == null && !isComposableRequired)

Reply:
Owing to disallowing to back to composable, i have to refactor these codes. 

---
In reply to: [268309133](https://github.com/OData/odata.net/pull/1400#discussion_r268309133) [](ancestors = 268309133)